python/cudf/dataframe/dataframe.py

```diff
@@ -2344,6 +2331,35 @@ def quantile(self,
                                          quant_index=False)
         return result
 
+    def select_dtypes(self, include=None):
+        """Return a subset of the DataFrame’s columns based on the column dtypes.
+
+        Parameters
+        ----------
+        include : [type]
+            [description] (the default is None, which [default_description])
+
+        """
+
+        # TODO: mechanism to hand int->int64 float->float64...
+
+        if isinstance(include, str):
+            include = [include]
+        df = DataFrame()
+
+        # convert any dtype() to string for comparison
+        include = [str(d) for d in include]
+
+        for x in self._cols.values():
+            if str(x.dtype) in include:
+                df.add_column(x.name, x.data)
+
+        return df
+
+    @property
+    def ndim(self):
+        return len(self.columns)
```

Comment:
I believe this is supposed to only return 2 for dataframes and 1 for series. See http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.ndim.html

Reply:
i'm going to remove `ndim` for now.  I like the way pandas does it by calling out to len(self.axes) and we don't currently have that.  I'll that build in a separate PR